### PR TITLE
Add fallback moveFromCache implementation

### PR DESCRIPTION
**Pull request by icewind1991**
For moving files from 3rdparty cache implementations.

### CODE REVIEW

**Comment by icewind1991** [comment]
cc @DeepDiver1975 @PVince81

**Comment by PVince81** [comment]
:+1:

File: tests/lib/files/cache/movefromcachetraittest.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?php
+/**
+ * Copyright (c) 2012 Robin Appelman <icewind@owncloud.com>
```
Comment (LukasReschke) [inline tests/lib/files/cache/movefromcachetraittest.php]: Old copyright header?

File: lib/private/files/cache/movefromcachetrait.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		$sourceCache->remove($sourcePath);
+	}
+
+	/**
+	 * Copy a file or folder in the cache
+	 *
+	 * @param \OCP\Files\Cache\ICache $sourceCache
+	 * @param ICacheEntry $sourceEntry
+	 * @param string $targetPath
+	 * @return bool|int
```
Comment (LukasReschke) [inline lib/private/files/cache/movefromcachetrait.php]: I don't see any return in the function?

File: lib/private/files/cache/movefromcachetrait.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	 */
+	abstract public function put($file, array $data);
+
+	/**
+	 * Move a file or folder in the cache
+	 *
+	 * @param \OCP\Files\Cache\ICache $sourceCache
+	 * @param string $sourcePath
+	 * @param string $targetPath
+	 * @return bool|int
```
Comment (LukasReschke) [inline lib/private/files/cache/movefromcachetrait.php]: I don't see any return in the function?

**Comment by icewind1991** [comment]
@LukasReschke: fixed

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**